plugin/completion.py

```diff
@@ -105,6 +113,47 @@ def get_text_edit_range(text_edit: Union[TextEdit, InsertReplaceEdit]) -> Range:
     return text_edit['range']
 
 
+def is_range(val: Optional[Any]) -> TypeGuard[Range]:
+    return isinstance(val, dict) and 'start' in val and 'end' in val
+
+
+def is_edit_range(val: Optional[Any]) -> TypeGuard[CompletionEditRange]:
+    return isinstance(val, dict) and 'insert' in val and 'replace' in val
+
+
+def completion_with_defaults(item: CompletionItem, item_defaults: CompletionItemDefaults) -> CompletionItem:
+    """ Currently supports defaults for: ["editRange", "insertTextFormat", "data"] """
+    default_text_edit = None  # type: Optional[Union[TextEdit, InsertReplaceEdit]]
+    edit_range = item_defaults.get('editRange')
+    if edit_range:
+        #  If textEditText is not provided and a list's default range is provided
+        # the label property is used as a text.
+        new_text = item.get('textEditText') or item['label']
+        if is_edit_range(edit_range):
+            default_text_edit = {
+                'newText': new_text,
+                'insert': edit_range.get('insert'),
+                'replace': edit_range.get('insert'),
+            }
+        elif is_range(edit_range):
+            default_text_edit = {
+                'newText': new_text,
+                'range': edit_range
+            }
+    if default_text_edit and 'textEdit' not in item:
+        item['textEdit'] = default_text_edit
+    default_insert_text_format = item_defaults.get('insertTextFormat')
+    if default_insert_text_format and 'insertTextFormat' not in item:
+        item['insertTextFormat'] = default_insert_text_format
+    default_data = item_defaults.get('data')
+    if default_data and 'data' not in item:
+        item['data'] = default_data
+    default_data = item_defaults.get('data')
+    if default_data:
+        item['data'] = item.get('data', default_data)
```

Comment:
Something went wrong here with suggestions.

```suggestion
    if default_data and 'data' not in item:
        item['data'] = default_data
```